Add a command-line flag for overriding pthreadpool size

dreiss opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47407 Make bundled inputs work with quantized zero inputs
* #46780 PyTorch NNAPI integration prototype
* **#46781 Add a command-line flag for overriding pthreadpool size**

Test Plan:
Passed it to speed_benchmark_torch and saw perf change.

Differential Revision: [D24752889](https://our.internmc.facebook.com/intern/diff/D24752889)

dreiss committed:
Add a command-line flag for overriding pthreadpool size

Test Plan:
Passed it to speed_benchmark_torch and saw perf change.

[ghstack-poisoned]

AshkanAliabadi commented on caffe2/utils/threadpool/ThreadPool.cc:
```diff
@@ -15,12 +15,18 @@ C10_DEFINE_int(caffe2_threadpool_android_cap, true, "");
   int numThreads = cpuinfo_get_processors_count();
 
+  if (FLAGS_pthreadpool_size) {
+    numThreads = FLAGS_pthreadpool_size;
```
Since caffe2_threadpool_android_cap and caffe2_threadpool_ios_cap are set to true by default, overriding the thread count at this point is still going to let those caps override this setting.  Is that the intended behavior?

dreiss replied:
Ah, no.  Good point.  I should move this down below that.  An explicit count should take priority.

dreiss committed:
Update on "Add a command-line flag for overriding pthreadpool size"


Test Plan:
Passed it to speed_benchmark_torch and saw perf change.

[ghstack-poisoned]
